src/Gateway/PayplugCreditCard.php

```diff
@@ -248,4 +247,111 @@ public function payment_fields()
 			$this->saved_payment_methods();
 		}
 	}
+
+	/**
+	 * Process the subscription scheduled payment
+	 */
+	public function scheduled_subscription_payment($amount, $order) {
+
+		$order_id      = PayplugWoocommerceHelper::is_pre_30() ? $order->id : $order->get_id();
+		$customer_id = PayplugWoocommerceHelper::is_pre_30() ? $order->customer_user : $order->get_customer_id();
+
+		$payplug_meta = wcs_get_subscription($order->get_meta('_subscription_renewal'))->get_parent()->get_meta("_payplug_metadata");
+
+		if (!$payplug_meta ) {
+			PayplugGateway::log('Could not find the payment token or the payment doesn\'t belong to the current user.', 'error');
+			throw new \Exception(__('Invalid payment method.', 'payplug'));
+		}
+
+		// Retrrienve the saved token
+
+		$token = $this->api->payment_retrieve($payplug_meta['transaction_id'])->card->id;
+
+
+		if (!$token || (int) $customer_id !== (int) $token->get_user_id()) {
```

Comment:
$token is a string something like `card_7Kxxxxxxxxxxxxx`, the `$token->get_user_id()` generates an error

Reply:
fixed